pages/index.tsx

```diff
@@ -20,18 +20,51 @@ import { StakingContractCode } from '../helpers/runkit';
 
 ChartJS.register([CategoryScale, LinearScale, PointElement, LineElement]);
 
+interface StorageProps {
+  newValue: string | number | undefined,
+  key: string,
+  floatValue: number,
+  currentValue: number | string
+}
+
 export default function Home() {
   const [totalDevelopers, setTotalDevelopers] = useState(0);
   const [inProgress, setInProgress] = useState(0)
-  const [beproStaked, setBeproStaked] = useState(0)
-  const [onNetwork, setOnNetwork] = useState(0)
+  const [beproStaked, setBeproStaked] = useState<number | string>(0)
+  const [onNetwork, setOnNetwork] = useState<number | string>(0)
+  const [totalAmount, setTotalAmount] = useState<number | string>(0);
   const { getTotalDevelopers } = useApi()
+
+
+  function handleLocalStorage({floatValue, newValue, key, currentValue}: StorageProps) {
+    if(localStorage.getItem(key) !== String(currentValue) && floatValue > 0){
+      localStorage.setItem(key, String(newValue))
+    }
+  }
  
   function initialize() {
-    BeproService._network.start().then(() => {
+    BeproService._network.start().then(async () => {
       BeproService.getOpenIssues().then(setInProgress);
-      BeproService.getTokensStaked().then(setOnNetwork);
-      BeproService.getBEPROStaked().then(setBeproStaked);
+      const tokens = await BeproService.getTokensStaked().then(value => {
```

Comment:
This can be reduced,

```ts
const tokens = await BeproService.getTokensStaked();
const oracles = await BeproService.getBEPROStaked();
const total = numberToUx(+tokens + +oracles);

localStorage.setItem('onNetwork', tokens);
localStorage.setItem('oracles', oracles);
localStorage.setItem('total', total);

setOnNetwork(tokens);
setBeproStaked(oracles);
setTotalAmount(total);
```